Cython/Compiler/ExprNodes.py

```diff
@@ -7047,6 +7029,32 @@ def analyse_as_type_attribute(self, env):
                     error(self.pos, "%s not a known value of %s" % (self.attribute, type))
         return None
 
+    def _create_ubcm_entry(self, entry, env):
+        # Create a temporary entry describing the unbound C method in `entry`
+        # as an ordinary function.
+        if entry.func_cname and not hasattr(entry.type, 'op_arg_struct'):
```

Comment:
I'm aware that you only copied this code, but `hasattr()` has a code smell. If we need to test whether this attribute exists in order to do something generic, maybe it should just always exist? And possibly just be `None` by default? Conditionally setting an attribute that would otherwise not exist breaks assumptions about object types/classes.

Reply:
I replaced the `hasattr()` with a check against `None`, and added a class attribute `op_args_struct` to `CFuncType`. I don't believe there any other instances of doing `hasattr(..., "op_arg_struct")`, and CI seems to have passed.